From b45f07bfbe74c333f1017810c2409e1aa6077a1b Mon Sep 17 00:00:00 2001
From: Szabolcs David <davidsz@inf.u-szeged.hu>
Date: Thu, 10 Jul 2014 05:13:41 -0700
Subject: <chromium> Add FirstVisuallyNonEmptyLayout support for
 RenderViewObserver

This is needed by the implementation of loadVisuallyCommitted in the QtWebEngine.

Change-Id: Ic0fcdc00f7963ea1088a1ae910bb2dbe26e9e12c
Reviewed-by: Jocelyn Turcotte <jocelyn.turcotte@digia.com>
---
 chromium/content/public/renderer/render_view_observer.h | 3 +++
 chromium/content/renderer/render_view_impl.cc           | 2 ++
 2 files changed, 5 insertions(+)

diff --git a/chromium/content/public/renderer/render_view_observer.h b/chromium/content/public/renderer/render_view_observer.h
index 9578867cbff..49e5c5c56d0 100644
--- a/chromium/content/public/renderer/render_view_observer.h
+++ b/chromium/content/public/renderer/render_view_observer.h
@@ -46,6 +46,9 @@ class RenderViewImpl;
 class CONTENT_EXPORT RenderViewObserver : public IPC::Listener,
                                           public IPC::Sender {
  public:
+
+  virtual void OnFirstVisuallyNonEmptyLayout() {}
+
   // By default, observers will be deleted when the RenderView goes away.  If
   // they want to outlive it, they can override this function.
   virtual void OnDestruct();
diff --git a/chromium/content/renderer/render_view_impl.cc b/chromium/content/renderer/render_view_impl.cc
index 134281769ca..de0a60a86d5 100644
--- a/chromium/content/renderer/render_view_impl.cc
+++ b/chromium/content/renderer/render_view_impl.cc
@@ -4105,6 +4105,8 @@ void RenderViewImpl::didFirstVisuallyNonEmptyLayout(WebFrame* frame) {
       InternalDocumentStateData::FromDataSource(frame->dataSource());
   data->set_did_first_visually_non_empty_layout(true);
 
+  FOR_EACH_OBSERVER(RenderViewObserver, observers_, OnFirstVisuallyNonEmptyLayout());
+
 #if defined(OS_ANDROID)
   // Update body background color if necessary.
   SkColor bg_color = webwidget_->backgroundColor();
-- 
cgit v1.2.1

